libcudacxx/include/cuda/std/__cccl/builtin.h

```diff
@@ -1245,4 +1246,91 @@
 #  define _CCCL_BUILTIN_STRLEN(...) __builtin_strlen(__VA_ARGS__)
 #endif
 
+// Some compilers provide std::move/std::forward/etc as builtins
+#if defined(__cplusplus) && !defined(__CUDA_ARCH__)
```

Comment:
Why is that needed?

Reply:
the `__CUDA_ARCH__` part? `::std::move`/`::std::forward`,et.al do not have `__device__` annotations.

Reply:
I was more about the `__cplusplus` part, but AFAIK nvcc treats at least `std::move` differently. I am 90% sure also forward